### PR TITLE
Create endpoint to return JSON release status

### LINKED ISSUE #70: publish latest successfully tested build via API
The purpose of this API would be to facilitate a way for other systems/projects/products to daisy-chain off of origin/openshift product CI flows, optimizing when and how often these external processes run.

/cc @ethernetdan

### PR DESCRIPTION
**Pull request by ethernetdan**
Resolves #70.

Not sure the best way to test this so recreating release objects in local cluster.

/cc @stevekuznetsov @jeremyeder

### CODE REVIEW

File: cmd/release-controller/http_helper.go [hunk 1]
```diff
@@ -16,22 +16,22 @@ import (
 )
 
 type ReleasePage struct {
-	BaseURL string
-	Streams []ReleaseStream
+	BaseURL string          `json:"baseURL"`
```
Comment (smarterclayton) [inline cmd/release-controller/http_helper.go]: I don't think we want these to be public APIs.

If the use case is "you want to get the latest release for a given stream", then the URL would be

```
/releasestream/{name}/latestaccepted
```

and the implementation should create a new object that has fields `name`, `pullSpec` and that's it.

Comment (ethernetdan) [reply cmd/release-controller/http_helper.go]: Makes sense, I’ll reduce it to those fields and change the route

**Comment by jeremyeder** [comment]
@smarterclayton is this ready for merge?

File: cmd/release-controller/http.go [hunk 2]
```diff
@@ -562,6 +564,51 @@ func (c *Controller) httpReleaseInfo(w http.ResponseWriter, req *http.Request) {
...
+
+	tags := tagsForRelease(r)
+	if len(tags) == 0 {
+		err = fmt.Errorf("no tags found for stream '%s'", streamName)
+
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	latestTag := tags[0]
```
Comment (smarterclayton) [inline cmd/release-controller/http.go]: Is this accepted?  I think you need to filter by accepted (`tagReferencesByPhase(Accepted)`)

**Comment by ktdreyer** [comment]
Is there any documentation about this REST API and how I would use it? How will users discover it?

**Comment by smarterclayton** [comment]
Folded into #78, added some tweaks (see the PR description).  Thanks!

**Pull request closed without merging**